src/data/set/basic.lean

```diff
@@ -240,6 +240,12 @@ theorem subset.rfl {s : set α} : s ⊆ s := subset.refl s
 @[trans] theorem mem_of_eq_of_mem {x y : α} {s : set α} (hx : x = y) (h : y ∈ s) : x ∈ s :=
 hx.symm ▸ h
 
+lemma ne_of_mem_nmem {x y : α} (hx : x ∈ s) (hy : y ∉ s) : y ≠ x :=
```

Comment:
We already have `ne_of_mem_of_not_mem`.

Reply:
Oh wow, thanks: way down in `logic/basic.lean` -- no wonder I missed it.

It's a bit disappointing that `library_search` couldn't find it either.